### PR TITLE
Loading data from netcdf is dependent on both in_vars and out_vars for ERA5

### LINKED ISSUE #50: out_vars in ERA5Forecasting module works only when out_vars is a subset of in_vars
**Describe the bug**
When calling init for [`ERA5Forecasting`](https://github.com/aditya-grover/climate-learn/blob/42915777b9f53475e7dfc6d463803e1f48c0e986/src/climate_learn/data/modules/era5_module.py#L98), the arguments passed to it's parent class are [`(root_dir, root_highres_dir, in_vars, years, split)`](https://github.com/aditya-grover/climate-learn/blob/42915777b9f53475e7dfc6d463803e1f48c0e986/src/climate_learn/data/modules/era5_module.py#L113). Because of this while creating the data_dict , only variables that are part of in_vars are loaded from netcdf files. See lines [45](https://github.com/aditya-grover/climate-learn/blob/42915777b9f53475e7dfc6d463803e1f48c0e986/src/climate_learn/data/modules/era5_module.py#L45) and [61](https://github.com/aditya-grover/climate-learn/blob/42915777b9f53475e7dfc6d463803e1f48c0e986/src/climate_learn/data/modules/era5_module.py#L61).

This can result in potential bug at line [122](https://github.com/aditya-grover/climate-learn/blob/42915777b9f53475e7dfc6d463803e1f48c0e986/src/climate_learn/data/modules/era5_module.py#L122) when we create the output data, if the output variables are not a subset of input variables.

### PR DESCRIPTION
**Pull request by prakhar6sharma**
Fixes #50.

### CODE REVIEW

**Comment by tung-nd** [comment]
Thanks Prakhar for the fix. Can you take a look at the new Iterable data module too?

**Review by jasonjewik — CHANGES_REQUESTED** [review]
Please add a test to show that your changes allow `out_vars` to NOT be a subset of `in_vars`. You can add your test as a new file in the top-level `tests/` directory. The file name must end with `_test.py`, _e.g._ `instantiate_era5_test.py`. Reference [instantiate_models_test.py](https://github.com/aditya-grover/climate-learn/blob/main/tests/instantiate_models_test.py) to see how to create a test fixture. More info can be found at the [pytest getting started guide](https://docs.pytest.org/en/7.1.x/getting-started.html).

To run your test locally, first install ClimateLearn's test dependencies. Assuming you are currently in the `climate-learn` directory:
```bash
pip install -e .[test]
```
Then, to run the tests:
```bash
pytest tests/
```
Let me know if you have further questions. I will eventually update the development handbook with a more thorough guide to testing.

**Comment by prakhar6sharma** [comment]
Will do that.

**Comment by prakhar6sharma** [comment]
Sorry for the delay, in the changes. I was a little busy over the past 2 days.

@jasonjewik for me to write a test that instantiates the ERA5Forecasting and ERA5Downscaling module, I would need to provide the data path as argument. Should I use the local path here for the test? if not then what should I do here?

**Comment by prakhar6sharma** [comment]
The tests currently work on the mint1 server because I have passed the absolute paths to the data directory. But they won't pass here because GitHub has no way of accessing the data.

**Comment by jasonjewik** [comment]
Hi @prakhar6sharma, I think we should use [the `tmp_path_factory` fixture](https://docs.pytest.org/en/latest/how-to/tmp_path.html#the-tmp-path-factory-fixture).

I can't checkout your branch (since it's on your own fork) and make the change myself. Can you implement this? In my head, the way this would work is that you create temp directories for all the dataset/variables and then download exactly one weatherbench file to each directory. It has to be the file corresponding to the earliest year in the range because of [these lines in the ERA5 module](https://github.com/aditya-grover/climate-learn/blob/9e5036d5af3180410fabf3af1341b91989146eea/src/climate_learn/data/modules/era5_module.py#L85-L86). E.g., for your tests, the corresponding weatherbench files would be for the year 2000.

**Comment by prakhar6sharma** [comment]
Downloading files by calling the `_download_weatherbench` is pretty slow. This is what I was doing when downloading for a single variable `download(root='/tmp', source='weatherbench', variable='2m_temperature', dataset = "era5", resolution = "5.625")`. Do you know if there is a way I can download only for a single year for a given variable.

**Comment by jasonjewik** [comment]
@prakhar6sharma IMO a unit test should not rely upon other functionalities of ClimateLearn. However, you're right, even if we download the files manually, WeatherBench only offers ZIP archives that contain all years. I would suggest downloading [ERA5 data from Copernicus](https://cds.climate.copernicus.eu/cdsapp#!/dataset/reanalysis-era5-single-levels?tab=overview), but that requires a CDS API key. I don't think we can install such a key in the GitHub test environment. So... I'm going to try to find a way to skip your tests when we're in the GitHub test environment.

**Comment by jasonjewik** [comment]
Hi @prakhar6sharma, please do this in your tests:
```python
import os
import pytest
# ... your other imports

GITHUB_ACTIONS = os.environ.get("GITHUB_ACTIONS") == "true"

@pytest.mark.skipif(GITHUB_ACTIONS, reason="only works locally")
def some_function_test(...):
    # ... your code here
```
This will allow GitHub to skip your tests.

See #62 for a full example.

**Comment by prakhar6sharma** [comment]
> Thanks Prakhar for the fix. Can you take a look at the new Iterable data module too?

I can draft another PR for this. Just want to get this one closed as there would be more and more merge conflicts if I delay this PR.

File: tests/data/era5_dataset_test.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import os
+from climate_learn.data.modules import ERA5
+import pytest
+
+DATA_PATH = "/data0/datasets/weatherbench/data/weatherbench/"
+
```
Comment (jasonjewik) [inline tests/data/era5_dataset_test.py:6]: Just a style thing: can we get rid of the line between `DATA_PATH` and `GITHUB_ACTIONS`? Or does black forbid it?

File: tests/data/era5_downscaling_dataset_test.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import os
+from climate_learn.data.modules import ERA5Downscaling
+from climate_learn.utils.datetime import Hours
+import pytest
+
+DATA_PATH = "/data0/datasets/weatherbench/data/weatherbench/"
+
```
Comment (jasonjewik) [inline tests/data/era5_downscaling_dataset_test.py:7]: Same as previous comment.

File: tests/data/era5_forecasting_dataset_test.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import os
+from climate_learn.data.modules import ERA5Forecasting
+from climate_learn.utils.datetime import Hours
+import pytest
+
+DATA_PATH = "/data0/datasets/weatherbench/data/weatherbench/"
+
```
Comment (jasonjewik) [inline tests/data/era5_forecasting_dataset_test.py:7]: Same as previous comment.

**Review by jasonjewik — CHANGES_REQUESTED** [review]

File: tests/data/era5_dataset_test.py (hunk 1 above)
Comment (prakhar6sharma) [reply tests/data/era5_dataset_test.py]: My bad. Black doesn't forbid it. I have removed the line.

File: tests/data/era5_downscaling_dataset_test.py (hunk 2 above)
Comment (prakhar6sharma) [reply tests/data/era5_downscaling_dataset_test.py]: See reply above.

File: tests/data/era5_forecasting_dataset_test.py (hunk 3 above)
Comment (prakhar6sharma) [reply tests/data/era5_forecasting_dataset_test.py]: See reply above.

**Review by jasonjewik — APPROVED** [review]
lgtm! approved ✅

**Pull request merged**